src/friends.rs

```diff
@@ -111,6 +111,31 @@ impl<Manager> Friends<Manager> {
             friends
         }
     }
+
+    /// Get a list of users from a specific source (e.g. gameserver)
+    /// Despite the misleading name, it does not return only friends
+    /// but all users from the source that are currently connected.
+    /// This does however require some local client to be connected to the source.
+    pub fn get_friends_from_source(&self, source: SteamId) -> Vec<Friend<Manager>> {
+        unsafe {
+            let count =
+                sys::SteamAPI_ISteamFriends_GetFriendCountFromSource(self.friends, source.0);
+            if count == -1 {
+                return Vec::new();
+            }
+            let mut friends = Vec::with_capacity(count as usize);
+            for idx in 0..count {
+                let friend = SteamId(sys::SteamAPI_ISteamFriends_GetFriendFromSourceByIndex(
```

Comment:
Should this be implemented as an iterator instead?

Reply:
the vec is initialized with size `count` to make it explicit that the index is important. 
the steam api requires the index to get further information

what do you think?

Reply:
I don't think that's the issue I'm raising here. Forcing this into Vec requires it to allocate on each call. If we had a lazy iterator, the caller can the n decide whether to collect it into a Vec or just iterate/filter/postprocess or whatever they please with it instead.